src/controller/java/AndroidCallbacks.cpp

```diff
@@ -273,25 +267,43 @@ void ReportCallback::OnAttributeData(const app::ConcreteDataAttributePath & aPat
                    ReportError(attributePathObj, nullptr, CHIP_JNI_ERROR_EXCEPTION_THROWN));
 #endif
     // Create TLV byte array to pass to Java layer
-    size_t bufferLen                  = readerForJavaTLV.GetRemainingLength() + readerForJavaTLV.GetLengthRead();
+    TLV::TLVReader readerForJavaTLV;
+    readerForJavaTLV.Init(*apData);
+    //
+    // Size of the start container and end container are just 1 byte each, but, let's just be safe.
+    //
+    size_t bufferLen                  = UINT32_MAX;
     std::unique_ptr<uint8_t[]> buffer = std::unique_ptr<uint8_t[]>(new uint8_t[bufferLen]);
     uint32_t size                     = 0;
-    // The TLVReader's read head is not pointing to the first element in the container, instead of the container itself, use
-    // a TLVWriter to get a TLV with a normalized TLV buffer (Wrapped with an anonymous tag, no extra "end of container" tag
-    // at the end.)
+
     TLV::TLVWriter writer;
     writer.Init(buffer.get(), bufferLen);
-    err = writer.CopyElement(TLV::AnonymousTag(), readerForJavaTLV);
+    TLV::TLVType outer;
+
+    err = writer.StartContainer(TLV::AnonymousTag(), TLV::kTLVType_Structure, outer);
+    VerifyOrReturn(err == CHIP_NO_ERROR, ReportError(attributePathObj, nullptr, err));
+    TLV::Tag tag;
+    err = ConvertTlvTag(aPath.mAttributeId, tag);
```

Comment:
Why does this format ever get into TLV? Why can't we sent the raw TLV and potentially decoded path IBs if that is needed since those are standard everywhere?

Reply:
Wanna keep the tlv and json data are exactly same for data, thanks.